app/src/main/java/com/example/recipeapp/fragments/RecipeSearchFragment.java

```diff
@@ -54,6 +60,7 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         client = new RecipeClient(getContext());
         recipes = new ArrayList<>();
         adapter = new RecipeSearchAdapter(getContext(), recipes);
+        getUser();
```

Comment:
So two things here:

- generally, you don't want to do I/O (networking or disk operations, etc.) from inside a constructor
- Android Fragment constructors are [weird](https://stackoverflow.com/questions/9245408/best-practice-for-instantiating-a-new-android-fragment).

The second item is a bit longer of a discussion. Put simply, we have to let the framework take care of making new fragments and then do our work inside of the [lifecycle callbacks](https://www.geeksforgeeks.org/fragment-lifecycle-in-android/). Typically, stuff like this should happen in `onCreate` instead of the constructor.